src/game.cpp

```diff
@@ -8513,17 +8513,22 @@ bool game::get_liquid_target( item &liquid, item *const source, const int radius
                               const monster *const source_mon,
                               liquid_dest_opt &target )
 {
-    if( !liquid.made_of( LIQUID ) ) {
+    if( item::find_type( liquid.typeId() )->phase == SOLID ) {
         dbg( D_ERROR ) << "game:handle_liquid: Tried to handle_liquid a non-liquid!";
         debugmsg( "Tried to handle_liquid a non-liquid!" );
         // "canceled by the user" because we *can* not handle it.
         return false;
     }
 
+    const std::string liquid_name = liquid.display_name( liquid.charges );
```

Comment:
Get the name *inside* the `if` block below. You only need it inside that block (which may not even be run at all).